sky/templates/azure-ray.yml.j2

```diff
@@ -87,17 +87,19 @@ setup_commands:
   - pip3 --version > /dev/null 2>&1 || (curl -sSL https://bootstrap.pypa.io/get-pip.py -o get-pip.py && python3 get-pip.py && echo "PATH=$HOME/.local/bin:$PATH" >> ~/.bashrc);
     (type -a python | grep -q python3) || echo 'alias python=python3' >> ~/.bashrc;
   - which conda > /dev/null 2>&1 || (wget -nc https://repo.anaconda.com/miniconda/Miniconda3-latest-Linux-x86_64.sh && bash Miniconda3-latest-Linux-x86_64.sh -b && eval "$(/home/azureuser/miniconda3/bin/conda shell.bash hook)" && conda init && conda config --set auto_activate_base true);
+  # patch the buggy ray files and enable `-o allow_other` option for `goofys`
+  # This also kills the service that is holding the lock on dpkg (problem only exists on aws/azure, not gcp)
   - (type -a pip | grep -q pip3) || echo 'alias pip=pip3' >> ~/.bashrc;
     (pip3 list | grep ray | grep {{ray_version}} 2>&1 > /dev/null || pip3 install -U ray[default]=={{ray_version}}) && mkdir -p ~/sky_workdir && mkdir -p ~/.sky/sky_app && touch ~/.sudo_as_admin_successful;
+    pip3 uninstall skypilot -y &> /dev/null;
+    pip3 install "$(echo {{sky_remote_path}}/skypilot-*.whl)[azure]";
+    python3 -c "from sky.skylet.ray_patches import patch; patch()";
     sudo systemctl stop unattended-upgrades;
     sudo kill -9 `sudo lsof /var/lib/dpkg/lock-frontend | awk '{print $2}' | tail -n 1` || true;
     sudo pkill -9 apt-get;
     sudo pkill -9 dpkg;
     sudo dpkg --configure -a;
-    sudo sed -i 's/#user_allow_other/user_allow_other/g' /etc/fuse.conf; # This is needed for `-o allow_other` option for `goofys`
-  - pip3 uninstall skypilot -y &> /dev/null;
-    pip3 install "$(echo {{sky_remote_path}}/skypilot-*.whl)[azure]";
-    python3 -c "from sky.skylet.ray_patches import patch; patch()" # patch the buggy ray files
+    sudo sed -i 's/#user_allow_other/user_allow_other/g' /etc/fuse.conf;
```

Comment:
Do we want to keep this section or at least this command the same?

E.g., in gcp we have
```
touch /etc/fuse.conf; sudo sed -i 's/#user_allow_other/user_allow_other/g' /etc/fuse.conf; # This is needed for `-o allow_other` option for `goofys`;
```
with the `touch` command and a comment. Here the comment is moved to the top and may not immediately remind the reader which cmd it's referring to.